src/components/OpenWith/OpenWith.js

```diff
@@ -86,16 +102,96 @@ class OpenWith extends PureComponent<Props> {
     }
 
     /**
-     * Fetches the list of potential integrations on load
      *
      * @private
      * @inheritdoc
      * @return {void}
      */
     componentDidMount() {
-        /* no-op */
+        const { fileId }: Props = this.props;
+        this.api.getOpenWithAPI(false).get({
+            id: fileId,
+            successCallback: this.fetchAppIntegrations,
+            errorCallback: this.fetchErrorCallback
+        });
     }
 
+    /**
+     * Fetch app intgrations info needed to render.
+     *
+     * @param {OpenWithIntegrations} openWithintegrations - The available Open With integrations
+     * @return {void}
+     */
+    fetchAppIntegrations = (openWithintegrations: OpenWithIntegrations) => {
+        const { items } = openWithintegrations;
+        const appIntegrationInfoPromises = [];
+
+        items.forEach((integrationItem: OpenWithIntegrationItem) => {
+            const { app_integration: { id } } = integrationItem;
+            const promise = this.fetchAppIntegrationPromise(id);
+            appIntegrationInfoPromises.push(promise);
+        });
+
+        // Update state after we have information for all integrations, which are required to render.
+        Promise.all(appIntegrationInfoPromises)
+            .then((integrations) => {
+                const openWithintegrationsWithIntegrationData = this.completeOpenWithIntegrationData(
+                    openWithintegrations,
+                    integrations
+                );
+
+                this.setState({ integrationsData: openWithintegrationsWithIntegrationData });
```

Comment:
Should we be using async/await for stuff like this? It seems like each element has its own strategy for async logic.